src/Sentry.Unity/Integrations/UnityLogHandlerIntegration.cs

```diff
@@ -1,54 +1,102 @@
+
+        public UnityLogHandlerIntegration(IApplication? application = null)
         {
             _application = application ?? ApplicationAdapter.Instance;
-            _eventCapture = eventCapture;
         }
 
         public void Register(IHub hub, SentryOptions sentryOptions)
         {
             _hub = hub;
             _sentryOptions = sentryOptions as SentryUnityOptions;
 
-            _application.LogMessageReceived += OnLogMessageReceived;
+            _unityLogHandler = Debug.unityLogger.logHandler;
+            Debug.unityLogger.logHandler = this;
+
             _application.Quitting += OnQuitting;
         }
 
-        // Internal for testability
-        internal void OnLogMessageReceived(string? condition, string? stackTrace, LogType type)
+        public void LogException(Exception exception, UnityEngine.Object context)
+        {
+            CaptureException(exception, context);
+            _unityLogHandler.LogException(exception, context);
+        }
+
+        internal void CaptureException(Exception exception, UnityEngine.Object? context)
         {
-            if (condition is null || _hub?.IsEnabled != true)
+            if (_hub?.IsEnabled is not true)
             {
                 return;
             }
 
-            if (condition.StartsWith(UnityLogger.LogPrefix, StringComparison.Ordinal))
+            // TODO: Capture the context (i.e. grab the name if != null)
+
+            // NOTE: This might not be entirely true, as a user could as well call `Debug.LogException`
+            // and expect a handled exception but it is not possible for us to differentiate
+            // https://docs.sentry.io/platforms/unity/troubleshooting/#unhandled-exceptions---debuglogexception
+            exception.Data[Mechanism.HandledKey] = false;
+            exception.Data[Mechanism.MechanismKey] = "Unity.LogException";
+            _ = _hub.CaptureException(exception);
+
+            // So the next event includes this error as a breadcrumb
+            _hub.AddBreadcrumb(message: $"{exception.GetType()}: {exception.Message}", category: "unity.logger", level: BreadcrumbLevel.Error);
+        }
+
+        public void LogFormat(LogType logType, UnityEngine.Object? context, string format, params object[] args)
+        {
+            CaptureLogFormat(logType, context, format, args);
+            _unityLogHandler.LogFormat(logType, context, format, args);
+        }
+
+        internal void CaptureLogFormat(LogType logType, UnityEngine.Object? context, string format, params object[] args)
+        {
+            if (_hub?.IsEnabled is not true)
             {
                 return;
             }
 
-            if (_sentryOptions?.EnableLogDebouncing == true)
+            // TODO: Figure out if format {0} and args.length == 1 is guaranteed?
+            // TODO: Capture the context (i.e. grab the name if != null)
+
+            if (!format.Equals("{0}") || args.Length is > 1 or <= 0)
```

Comment:
This will protect us from the uncertainty indicated by the TODO comment on line 77.

```suggestion
            if (args.Length == 0 || !format.Contains("{0}"))
```